binding-hs/src/Agent.hs

```diff
@@ -0,0 +1,71 @@
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE OverloadedStrings     #-}
+
+module Agent
+  ( runAgent
+  ) where
+
+import           Client
+import           Control.Monad              (replicateM_, when)
+import           Control.Monad.Loops        (whileM_)
+import           Control.Monad.Trans.Except (runExceptT)
+import           Data.Aeson                 (encode)
+import           Network.HTTP.Client        (Manager, defaultManagerSettings,
+                                             newManager)
+import           Servant.Client             (BaseUrl (..), Scheme (Http))
```

Comment:
It's a bit weird to be pulling in Servant as a required import from a dependency. Can we abstract it away and also export some defaults so that people don't need to do anything extra?